nequip/data/AtomicData.py

```diff
@@ -117,11 +117,19 @@ def _process_dict(kwargs, ignore_fields=[]):
             # Any property used as an index must be long (or byte or bool, but those are not relevant for atomic scale systems)
             # int32 would pass later checks, but is actually disallowed by torch
             kwargs[k] = torch.as_tensor(v, dtype=torch.long)
+        elif isinstance(v, bool):
+            kwargs[k] = torch.as_tensor(v)
         elif isinstance(v, np.ndarray):
             if np.issubdtype(v.dtype, np.floating):
                 kwargs[k] = torch.as_tensor(v, dtype=torch.get_default_dtype())
             else:
                 kwargs[k] = torch.as_tensor(v)
+        elif isinstance(v, list):
+            ele_dtype = np.array(v).dtype
+            if np.issubdtype(ele_dtype, np.floating):
+                kwargs[k] = torch.as_tensor(v, dtype=torch.get_default_dtype())
+            else:
+                kwargs[k] = torch.as_tensor(v)
```

Comment:
@Linux-cpp-lisp  could you check whether this is what you meant?

Reply:
yep, looks right :+1: